File: frontend/src/utils/api/airports.ts
import { API_BASE_URL } from "./config";

type Airport = {
  country_code: string;
  icao_code: string;
  label: string;
  lat: number;
  lon: number;
};

const fetchAirports = async (status?: number): Promise<Airport[]> => {
  try {
    const query = status !== undefined ? `?status=${status}` : "";
    const response = await fetch(`${API_BASE_URL}/v1/airports${query}`, {
      headers: {
        Accept: "application/json",
      },
    });

    if (!response.ok) {
      throw new Error("Failed to fetch airports");
    }

    const data: Airport[] = await response.json();
    return data;
  } catch (error) {
    console.error("Error fetching airports:", error);
    return [];
  }
};

export type { Airport };
export { fetchAirports };

File: backend/spec/factories/airports.rb
FactoryBot.define do
  factory :airport do
    country_code { "JP" }
    sequence(:icao_code) { |n| "TEST#{n}" }
    sequence(:label) { |n| "TEST AIRPORT#{n}" }
    sequence(:lat) { |n| 35.0 + n * 0.05 }
    sequence(:lon) { |n| 139.0 + n * 0.03 }
    uri { "http://example.com/test" }
    status { 1 }
  end
end
